query_optimizer/rules/ExtractCommonSubexpression.hpp

```diff
@@ -111,7 +111,7 @@ class ExtractCommonSubexpression : public Rule<physical::Physical> {
   bool visitAndCount(
       const expressions::ExpressionPtr &expression,
       ScalarCounter *counter,
-      ScalarHashable *hashable);
+      ScalarHashable *hashable) const;
```

Comment:
These three methods involve the invoking of `optimizer_context_->nextExprId()` which modifies  the context (symbol id counter) of `optimizer_context_`.